src/transformers/trainer.py

```diff
@@ -1019,6 +1019,8 @@ def train(
             resume_from_checkpoint = get_last_checkpoint(args.output_dir)
             if resume_from_checkpoint is None:
                 raise ValueError(f"No valid checkpoint found in output directory ({args.output_dir})")
+        elif resume_from_checkpoint is None:
+            resume_from_checkpoint = args.resume_from_checkpoint
```

Comment:
Won't this create ambiguity? Basically now a user may pass something to `train(resume_from_checkpoint=x)` and at the same time set  `args.resume_from_checkpoint=y`, so you have 2 ways to accomplish the same but it's not even obvious which one is superior to the other.

The added doc did say that `args.resume_from_checkpoint` acts as a default, but it's just very odd to have a default value here. But if one thinks hard it becomes clear that `train`'s arg with the same name overrides the `TrainingArguments`. It's just far from easy to grasp and can lead to confusing at training time. To me if I change `--resume_from_checkpoint` under pressure I'd expect the program to use that checkpoint and I won't remember that I may have another setting elsewhere. I struggle with this proposal.

Could we do this only for the example scripts and leave the Trainer unmodified? So that `--resume_from_checkpoint` is used explicitly if it's passed to the script `train(resume_from_checkpoint=args.resume_from_checkpoint)` if set.

i.e. the proposed new argument is not a `TrainingArguments` but just a helper for the scripts? like max samples?